sdk/core/System.ClientModel/src/Options/ClientPipelineOptions.cs

```diff
@@ -177,4 +220,47 @@ protected void AssertNotFrozen()
             throw new InvalidOperationException("Cannot change a ClientPipelineOptions instance after it has been used to create a ClientPipeline.");
         }
     }
+
+    #region Helpers
+
+    internal HttpClientPipelineTransport GetHttpClientPipelineTransport()
+    {
+        // If client-wide logging is enabled via ILogger, the shared client pipeline transport instance cannot be used. The transport will need the
+        // logger factory instance from the logging options.
+        // If client-wide logging is disabled, the shared client pipeline transport instance cannot be used. By default, client-wide logging is
+        // enabled.
+        if (_loggingOptions != null && (_loggingOptions.ClientWideLoggingIsDisabled || _loggingOptions.ClientWideLoggingIsEnabledViaILogger))
```

Comment:
Just to reduce complexity of the decision here (perhaps the criteria get more complex over time) should we introduce something like we did for `AddDefaultMessageLoggingPolicy` such as `UseSharedHttpClientPipelineTransport`?